File: extensions/maven/test/fixtures/projects/junit-java/src/test/java/com/example/SampleTest.java
package com.example;

import org.junit.jupiter.api.*;
import java.util.List;
import static org.junit.jupiter.api.Assertions.*;

public class SampleTest {

    @Test
    void plainPasses() { assertTrue(true); }

    @Test
    void plainFails() { assertEquals(1, 2, "plain fails"); }

    @TestFactory
    List<DynamicTest> dynamicTests() {
        return List.of(
            DynamicTest.dynamicTest("dynamic one", () -> assertTrue(true)),
            DynamicTest.dynamicTest("dynamic two", () -> fail("dynamic fails"))
        );
    }

    @Nested
    class InnerContext {
        @Test
        void nestedPasses() { assertTrue(true); }

        @Test
        void nestedFails() { fail("nested fails"); }
    }
}

File: extensions/maven/test/fixtures/projects/junit-kotlin/src/test/kotlin/com/example/SampleTest.kt
package com.example

import org.junit.jupiter.api.*
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertTrue

class SampleTest {

    @Test
    fun `plain passes`() { assertTrue(true) }

    @Test
    fun `plain fails`() { assertEquals(1, 2) }

    @TestFactory
    fun `dynamic tests`(): List<DynamicTest> = listOf(
        DynamicTest.dynamicTest("dynamic one") { assertTrue(true) },
        DynamicTest.dynamicTest("dynamic two") { throw AssertionError("dynamic fails") }
    )

    @Nested
    inner class `inner context` {
        @Test
        fun `nested passes`() { assertTrue(true) }

        @Test
        fun `nested fails`() { throw AssertionError("nested fails") }
    }
}
